File: app/src/main/java/com/martinlaizg/geofind/config/Preferences.java
package com.martinlaizg.geofind.config;

import android.content.SharedPreferences;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.martinlaizg.geofind.data.access.api.entities.Login;
import com.martinlaizg.geofind.data.access.database.entities.User;
import com.martinlaizg.geofind.utils.DateUtils;

public class Preferences {

	public static final String USER = "user";
	public static final String LOGIN = "login";

	public static User getLoggedUser(SharedPreferences sp) {
		String user_string = sp.getString(USER, "");
		Gson gson = new GsonBuilder().setDateFormat(DateUtils.DATE_FORMAT).create();
		return gson.fromJson(user_string, User.class);
	}

	public static void setLoggedUser(SharedPreferences sp, User user) {
		String user_string = user.toJson();
		sp.edit().putString(USER, user_string).apply();
	}

	public static void logout(SharedPreferences sp) {
		sp.edit().putString(USER, null).apply();
		sp.edit().putString(LOGIN, null).apply();
	}

	public static void setLogin(SharedPreferences sp, Login login) {
		String login_string = login.toJson();
		sp.edit().putString(LOGIN, login_string).apply();
	}

	public static Login getLogin(SharedPreferences sp) {
		String login_string = sp.getString(LOGIN, "");
		Gson gson = new GsonBuilder().setDateFormat(DateUtils.DATE_FORMAT).create();
		return gson.fromJson(login_string, Login.class);
	}
}

File: app/src/main/java/com/martinlaizg/geofind/config/Preferences.kt
package com.martinlaizg.geofind.config

import android.content.SharedPreferences
import com.google.gson.GsonBuilder
import com.martinlaizg.geofind.data.access.api.entities.Login
import com.martinlaizg.geofind.data.access.database.entities.User
import com.martinlaizg.geofind.utils.DateUtils

object Preferences {
	const val USER = "user"
	const val TOKEN = "token"
	private const val LOGIN = "login"
	fun getLoggedUser(sp: SharedPreferences?): User {
		val userString = sp!!.getString(USER, "")
		val gson = GsonBuilder().setDateFormat(DateUtils.DATE_FORMAT).create()
		return gson.fromJson(userString, User::class.java)
	}

	fun setLoggedUser(sp: SharedPreferences?, user: User) {
		val userString = user.toJson()
		sp!!.edit().putString(USER, userString).apply()
	}

	fun logout(sp: SharedPreferences) {
		sp.edit().putString(USER, null).apply()
		sp.edit().putString(LOGIN, null).apply()
	}

	fun setLogin(sp: SharedPreferences?, login: Login?) {
		val loginString = login!!.toJson()
		sp!!.edit().putString(LOGIN, loginString).apply()
	}

	fun getLogin(sp: SharedPreferences?): Login {
		val loginString = sp!!.getString(LOGIN, "")
		val gson = GsonBuilder().setDateFormat(DateUtils.DATE_FORMAT).create()
		return gson.fromJson(loginString, Login::class.java)
	}

	fun setToken(sp: SharedPreferences?, token: String?) {
		sp!!.edit().putString(TOKEN, token).apply()
	}

	fun getToken(sp: SharedPreferences): String? {
		return sp.getString(TOKEN, "")
	}
}